packages/core/src/browser/widgets/enhanced-preview-widget.ts

```diff
@@ -0,0 +1,27 @@
+// *****************************************************************************
+// Copyright (C) 2023 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { isFunction, isObject } from '../../common';
+
+export interface EnhancedPreviewWidget {
+    getEnhancedPreviewNode(): Node | undefined;
```

Comment:
Can this be any node or does it have to be a HTMLELement``?

Reply:
We use it here: https://github.com/eclipse-theia/theia/blob/bab13b9505905c40757c17bdfb7e95932d72b30d/packages/core/src/browser/shell/tab-bars.ts#L512C15-L512C15
`appendChild` accepts any `Node`: https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild

Reply:
I know that the typing works out, the question is: does it make sense to ever return a node that's not a HTMLElement?

Reply:
`document.createTextNode()` is not an `HTMLElement` as far as I know. I don't see why we should restrict this. 
Moreover it is consistent with the visual previews interface which uses `Node | undefined` as well.